### PR TITLE
Build SiteExtensions

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Addresses #11186 

WIP since it doesn't work yet.

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
Build worked locally, testing on CI now. The implementation is a little rough but I want to see if I can remove the need for a build script and use project dependencies instead. Also need to verify the produced site extensions and see if they are correct.

**Review by dougbu — APPROVED** [review]
Checks are passing and this looks good despite the need for a new script.

If you can't do it in this PR, please file a cleanup issue to expand the names of the `.binlog` files, they're too abbreviated for most readers and the names could be much longer.

**Review by dougbu — COMMENTED** [review]
@JunTaoLuo if you're making changes, please expand the binlog filenames e.g. SiteExtensions-LoggingBranch-x64.binlog

**Comment by JunTaoLuo** [comment]
Verifying the contents of the site extension packages. The only issue is the missing ancm folder in AspNetCoreRuntime.3.0.RID's content. Investigating.

**Comment by jkotalik** [comment]
@JunTaoLuo let me know if you need help with this.

File: src/SiteExtensions/build.cmd [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+)
+
+ECHO Building Microsoft.AspNetCore.AzureAppServices.SiteExtension
+CALL %RepoRoot%\build.cmd -forceCoreMsbuild -projects %~dp0LoggingAggregate\src\Microsoft.AspNetCore.AzureAppServices.SiteExtension\Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj /bl:artifacts/log/SiteExtensions-LoggingAggregate.binlog %*
+
+IF %ERRORLEVEL% NEQ 0 (
+   EXIT /b %ErrorLevel%
+)
+
+ECHO SiteExtensions successly built!
```
Comment (jkotalik) [inline src/SiteExtensions/build.cmd]:
```suggestion
ECHO SiteExtensions successfully built!
```

**Review by jkotalik — APPROVED** [review]
I don't want to block this as it required for preview7, but there are at least a few issues:
- Building site extensions takes 23 minutes
- Calling explicit build steps like this seems likely to break in the future.

I'd say we file a follow-up issue for this and unblock site extension builds for now.

Also, maybe we can consider making it so this only builds on internal ci?

**Comment by JunTaoLuo** [comment]
Sure we can optimize the build later. I compared against some pre-arcade builds and it seems like the difference in build time comes from the SiteExtension builds. https://github.com/aspnet/AspNetCore/issues/11534

**Pull request merged**